### ISSUE TITLE
Request promise not resolved on iOS

### ISSUE THREAD

**Issue by elkinjosetm** [zoontek/react-native-permissions#813]
### Bug summary

Requesting location permission on iOS is never resolving the promise, however, it is actually triggering the native API to request the permission, and the permission can get granted, but the promise is never resolved.

Using `check` somewhere else return the actual granted status.

I haven't tried any different location permission as than `PERMISSIONS.IOS.LOCATION_WHEN_IN_USE`, and we don't need to request high accuracy.

### Library version

3.9.3

### Environment info

```shell
System:
  OS: macOS 14.0
  CPU: (8) arm64 Apple M1
  Memory: 63.23 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.17.0
    path: ~/.nvm/versions/node/v18.17.0/bin/node
  Yarn:
    version: 3.6.1
    path: ~/.nvm/versions/node/v18.17.0/bin/yarn
  npm:
    version: 9.6.7
    path: ~/.nvm/versions/node/v18.17.0/bin/npm
  Watchman:
    version: 2023.07.10.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.13.0
    path: /Users/etorres/.rvm/gems/ruby-3.0.0/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.0
      - iOS 17.0
      - macOS 14.0
      - tvOS 17.0
      - watchOS 10.0
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10811636
  Xcode:
    version: 15.0/15A240d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.17
    path: /usr/bin/javac
  Ruby:
    version: 3.0.0
    path: /Users/etorres/.rvm/rubies/ruby-3.0.0/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.5
    wanted: 0.72.5
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
```

### Steps to reproduce

1. Request location permission on iOS

### Reproducible sample code

```js
const result = await request(PERMISSIONS.IOS.LOCATION_WHEN_IN_USE);
```

**Comment by zoontek** [comment]
@elkinjosetm Does this occur on 3.9.2?

**Comment by elkinjosetm** [comment]
@zoontek, yesterday I upgraded from 3.8.4 over to 3.9.3, and updated the podfile like it's described in the release notes. It was not happening on 3.8.4, but I haven't tried any other version in between.

I'll try 3.9.2 and let you know...

**Comment by elkinjosetm** [comment]
@zoontek I just tried 3.9.2, and can confirm that is working as expected, so it was introduced on 3.9.3.

**Comment by algera** [comment]
Same here, pinning to 3.9.2 seems to work OK, 3.9.3 may have introduced the issue.

**Comment by zoontek** [comment]
I will rollback tomorrow.

**Comment by zoontek** [comment]
Fixed in [3.10.0](https://github.com/zoontek/react-native-permissions/releases/tag/3.10.0)
